sites/docs/src/content/docs/tutorials/tests_and_test_data/components/03_testing_modules.md

```diff
@@ -0,0 +1,810 @@
+│ Load .nf-test/plugins/nft-compress/0.1.0/nft-compress-0.1.0.jar                                        │
+│ Load .nf-test/plugins/nft-vcf/1.0.7/nft-vcf-1.0.7.jar                                                  │
+│ Load .nf-test/plugins/nft-csv/0.1.0/nft-csv-0.1.0.jar                                                  │
+│ Load .nf-test/plugins/nft-utils/0.0.3/nft-utils-0.0.3.jar                                              │
+│ Load .nf-test/plugins/nft-fastq/0.0.1/nft-fastq-0.0.1.jar                                              │
+│ Load .nf-test/plugins/nft-anndata/0.1.0/nft-anndata-0.1.0.jar                                          │
+│                                                                                                        │
+│ Test Process BEDTOOLS_BAMTOBED                                                                         │
+│                                                                                                        │
+│   Test [824188d1] 'sarscov2 - bam' PASSED (3.277s)                                                     │
+│   Test [f4f6429b] 'stub' PASSED (3.161s)                                                               │
+│                                                                                                        │
+│                                                                                                        │
+│ SUCCESS: Executed 2 tests in 6.446s                                                                    │
+│                                                                                                        │
+╰────────────────────────────────────────────────────────────────────────────────────────────────────────╯
+INFO     All tests passed!
+```
+
+### Stub mode
+
+All nf-core modules require a `stub:` section to allow 'dry run'-like functionality for pipelines.
+This is something that we always want to test for.
+
+Furthermore, in some cases the module will produce output data that is too big for GitHub actions nodes, or the tests take too long or require too much resources.
+Therefore, the only option is to test the module in `stub` mode.
+
+Therefore when adding a stub test, you need to tell a test to run in stub mode by adding the `-stub` option as follows:
+
+```main.nf.test
+process "MODULE"
+config "./nextflow.config"
+
+
+test("custom evalue") {
+  options "-stub"
+    when {
+        process {
+            // test implementation
+        }
+    }
+}
+```
+
+:::note
+For modules that can never run full tests due to data being too large or requiring too much resources, this option can alternatively be added at the top of `main.nf.test` to have all tests run in stub mode.
+:::
+
+## Testing chained modules
+
+In some cases, rather than directly linking to pre-made test-data files, it may make sense to run an 'upstream module' in your test to output the required inputs of the module you want to test.
```

Comment:
```suggestion
In some cases, rather than directly linking to pre-made test-data files within the `when:` scope, it may make sense to run an 'upstream module' in your test that produces the required inputs of the module you want to test.
```